src/test/java/org/apache/commons/lang3/SystemUtilsTest.java

```diff
@@ -373,6 +373,33 @@ public void testOSMatchesNameAndVersion() {
     }
 
     @Test
+    public void testMatchOSMajorMinorVersion() {
+        String osName = "Mac OS X";
+        String osVersion = null;
+        assertFalse(SystemUtils.isOSMajorMinorVersionMatch(osName, osVersion, "Mac OS X", "10.10"));
+        osVersion = "10.10";
+        assertTrue(SystemUtils.isOSMajorMinorVersionMatch(osName, osVersion, "Mac OS X", "10.10"));
+        assertFalse(SystemUtils.isOSMajorMinorVersionMatch(osName, osVersion, "Mac OS X", "10.1"));
+        osVersion = "10.1";
+        assertFalse(SystemUtils.isOSMajorMinorVersionMatch(osName, osVersion, "Mac OS X", "10.10"));
+        assertTrue(SystemUtils.isOSMajorMinorVersionMatch(osName, osVersion, "Mac OS X", "10.1"));
+        osVersion = "10.10.1";
+        assertTrue(SystemUtils.isOSMajorMinorVersionMatch(osName, osVersion, "Mac OS X", "10.10"));
+        osVersion = "10.10.10";
+        assertTrue(SystemUtils.isOSMajorMinorVersionMatch(osName, osVersion, "Mac OS X", "10.10"));
+    }
+
+
+    @Test
+    public void testNotConfuseMacOSXPumaAndYosemite() {
```

Comment:
I don't understand why we need this test case. It won't do anything on most machines.